core/src/layers/complete.rs

```diff
@@ -188,7 +199,7 @@ impl<A: Access> CompleteAccessor<A> {
                 ));
             }
 
-            return Ok(RpStat::new(Metadata::new(EntryMode::DIR)));
+            return Ok(RpStat::new(meta));
```

Comment:
Let's keep this change but remove the `capability` related.

Thank you for the catch!